pytorch_pfn_extras/writing.py

```diff
@@ -361,11 +363,21 @@ class ThreadWriter(StandardWriter):
     def __init__(self, savefun=torch.save, fs=None, out_dir=None, **kwds):
         super().__init__(savefun=savefun, fs=fs, out_dir=out_dir, **kwds)
 
+    def _save_with_exitcode(self, *args, **kwargs):
+        try:
+            self.save(*args, **kwargs)
+        except Exception as e:
+            thread = threading.current_thread()
+            thread.exitcode = -1
+            print(
+                f'Error: ThreadWriter failed in thread "{thread.name}": '
+                f'{type(e).__name__}: {str(e)}')
```

Comment:
How about printing in stderr?
```suggestion
                f'{type(e).__name__}: {str(e)}', file=sys.stderr)
```

Reply:
Sounds nice, fixed!